### PR TITLE
Add Blackwell-related methods to CudaComputeCapability class

### PR DESCRIPTION
**Pull request by sergey-kozub**
Also remove unused methods `GetMaxResidentBlocksPerSM` and `GetMaxResidentWarpsPerSM`

### CODE REVIEW

**Review by akuegel — COMMENTED** [review]
I am not 100% sure whether this would count as a NFC. If possible, could you potentially add a test that calls the new IsAtLeastBlackwell method?

**Comment by sergey-kozub** [comment]
> could you potentially add a test that calls the new IsAtLeastBlackwell method?

I don't see any existing tests for `IsAtLeast(Volta|Ampere|Hopper)` methods.
Should I add those as well? Also, which test file should I put this to?

On another note, should we completely remove `GetMaxResidentBlocksPerSM` and `GetMaxResidentWarpsPerSM` methods, that are not used anywhere? They just add maintenance burden.

**Comment by akuegel** [comment]
> > could you potentially add a test that calls the new IsAtLeastBlackwell method?
> 
> I don't see any existing tests for `IsAtLeast(Volta|Ampere|Hopper)` methods. Should I add those as well? Also, which test file should I put this to?

Yes, historically we were not so strict about tests, but now there is a presubmit check. Please put the test file in the same directory. Adding calls for the other IsAtLeast methods would be very nice, but I would also approve without, as they are not new.

> On another note, should we completely remove `GetMaxResidentBlocksPerSM` and `GetMaxResidentWarpsPerSM` methods, that are not used anywhere? They just add maintenance burden.

Completely agree, let's remove them. I checked that also internally they are marked as dead code.

**Comment by sergey-kozub** [comment]
PTAL

**Review by akuegel — APPROVED** [review]
Thank you very much!

File: xla/stream_executor/device_description_test.cc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "xla/stream_executor/device_description.h"
+
+#include "tsl/platform/test.h"
```
Comment (akuegel) [inline xla/stream_executor/device_description_test.cc:17]: It seems the build target is missing a dependency for this header include

Comment (sergey-kozub) [reply xla/stream_executor/device_description_test.cc:17]: Fixed.

**Review by akuegel — APPROVED** [review]

**Pull request closed without merging**